v2.0/src/Components/Pages/ActionsPage/OneActionPage.js

```diff
@@ -37,22 +37,42 @@ class OneActionPage extends React.Component {
       showTestimonialLink: false,
       numberToShow: 3,
       tab: "description",
-      question: null
+      question: null,
+      action: null,
     };
     this.handleChange = this.handleChange.bind(this);
   }
+
   componentDidMount() {
     window.addEventListener("resize", this.chooseFontSize);
+
+    console.log("componentDidMount");
+    const { id } = this.props.match.params;
+    this.fetch(id)
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.match.params !== prevProps.match.params) {
+      const { id } = this.props.match.params;
+      this.fetch(id);
+    }
+  }
+
+  fetch(id) {
```

Comment:
i think it would be better to fetch the data from a action/reducer rather than in file